This patch fixes a case where it find crypt defined in libc (musl) but
not in specified libraries then it ends up assigning

LIBCRYPT="-l" which then goes into makefile cause all sort of problems
e.g.

ld: cannot find -l-m32
| collect2: error: ld returned 1 exit status

The reason is that -l appears on commandline with out any library and
compiler treats the next argument as library name whatever it is.


Upstream-Status: Pending
Signed-off-by: Khem Raj <raj.khem@gmail.com>

Index: Linux-PAM-1.1.6/configure.in
===================================================================
--- Linux-PAM-1.1.6.org/configure.ac
+++ Linux-PAM-1.1.6/configure.ac
@@ -400,7 +400,9 @@ AS_IF([test "x$ac_cv_header_xcrypt_h" =
   [crypt_libs="crypt"])
 
 BACKUP_LIBS=$LIBS
-AC_SEARCH_LIBS([crypt],[$crypt_libs], LIBCRYPT="-l$ac_lib", LIBCRYPT="")
+AC_SEARCH_LIBS([crypt],[$crypt_libs],
+	       [test "$ac_cv_search_crypt" = "none required" ||
+		LIBCRYPT="$ac_cv_search_crypt"])
 AC_CHECK_FUNCS(crypt_r crypt_gensalt_r)
 LIBS=$BACKUP_LIBS
 AC_SUBST(LIBCRYPT)
